.github/workflows/verify-python-compatibility.yml

```
@@ -82,3 +83,9 @@ jobs:
           output-to: 'step-summary'
           list-tests: 'failed'
           list-suites: 'failed'
+      - name: Upload artifact
+        if: failure()
+        uses: actions/upload-artifact@v3
+        with:
+          name: behavex_output
```

If there are multiple failures, it will overwrite the uploaded files so only the last one will remain.
This might solve this issue (I didnt test it)
```suggestion
          name: ${{ matrix.platform.name }}_behavex_output
```

---

Wouldn't you also have to rename the output on the source side?

---

Good point, added an id prefix (to avoid spaces in the names) in b63e14955e800e97b33517fd96635abeec98a341 and tested it in https://github.com/lordgamez/nifi-minifi-cpp/actions/runs/7475211172, seems to be working.